idea_town/frontend/static-src/app/views/experiment-page.js

```diff
@@ -156,24 +156,43 @@ export default PageView.extend({
     }
 
     app.webChannel.sendMessage(eventType, {
-      addon_id: this.model.addon_id,
-      xpi_url: this.model.xpi_url
+      addon_id: model.addon_id,
+      xpi_url: model.xpi_url
     });
-
-    // TODO:(DJ) need to setup some databinding and progress ui Since
-    // the addon can fail on install.
-    // https://github.com/mozilla/idea-town/issues/199
-    this.model.enabled = !this.model.enabled;
   },
 
   install(evt) {
     evt.preventDefault();
-    this._updateAddon(true);
+    const width = evt.target.offsetWidth;
+    evt.target.style.width = width + 'px';
+    evt.target.classList.add('state-change');
+
+    this.updateAddon(true, this.model);
+    app.on('webChannel:addon-install:install-ended', () => {
+      this.model.enabled = !this.model.enabled;
+      evt.target.classList.remove('state-change');
+    });
   },
 
-  uninstall(evt) {
+  uninstall(cb, model) {
+    const uninstallButton = document.getElementById('uninstall-button');
+    const feedbackButton = document.getElementById('feedback-button');
+    const width = uninstallButton.offsetWidth;
+    uninstallButton.style.width = width + 'px';
+    uninstallButton.classList.add('state-change');
+    feedbackButton.style.display = 'none';
```

Comment:
Hmm, I don't think we need to mess with feedback button styles here. [We have these hooked up with data binding to the `enabled` prop of the experiment, no?](https://github.com/mozilla/idea-town/blob/master/idea_town/frontend/static-src/app/views/experiment-page.js#L36) When `enabled` get set to false, the feedback button should toggle to invisible. Also, I'm seeing an issue where the "give feedback" button doesn't actually go away on experiment disable - not sure if that's related to this.